sdk/digitaltwins/Azure.DigitalTwins.Core/src/DigitalTwinsClient.cs

```diff
@@ -233,9 +258,15 @@ public virtual Task<Response<string>> CreateDigitalTwinAsync(string digitalTwinI
         /// </exception>
         /// The exception that captures the errors from the service. Check the <see cref="RequestFailedException.ErrorCode"/> and <see cref="RequestFailedException.Status"/> properties for more details.
         /// </exception>
-        public virtual Response<string> CreateDigitalTwin(string digitalTwinId, string digitalTwin, CreateDigitalTwinOptions options = null, CancellationToken cancellationToken = default)
+        public virtual Response<T> CreateDigitalTwin<T>(string digitalTwinId, string digitalTwin, CreateDigitalTwinOptions options = null, CancellationToken cancellationToken = default)
         {
-            return _dtRestClient.Add(digitalTwinId, digitalTwin, options, cancellationToken);
+            // Get the digital twin as a Stream object
+            Response<Stream> digitalTwinStream = _dtRestClient.Add(digitalTwinId, digitalTwin, options, cancellationToken);
+
+            // Deserialize the stream into the generic type
+            T deserializedDigitalTwin = (T)_objectSerializer.Deserialize(digitalTwinStream, typeof(T), cancellationToken);
+
+            return Response.FromValue<T>(deserializedDigitalTwin, digitalTwinStream.GetRawResponse());
```

Comment:
This logic looks correct to me. I'd like to see the samples/tests make use of deserializing into "BasicDigitalTwin" and "BasicRelationship" rather than into "object" though

Reply:
Yup. I am currently changing all <object>s into proper defined types.